doc/lsst.pipe.base/task-framework-overview.rst

```diff
@@ -0,0 +1,49 @@
+.. NOTE: this topic should be edited into the modernized topic-based documentation style.
+
+.. _task-framework-overview:
+
+##############################
+Overview of the task framework
+##############################
+
+:doc:`lsst.pipe.base <index>` provides a data processing pipeline infrastructure.
+Data processing is performed by **tasks**, which are instances of `lsst.pipe.base.Task` or `lsst.pipe.base.CmdLineTask`.
+Tasks perform a wide range of data processing operations, from basic operations such as assembling raw images into CCD images (trimming overscan), fitting a WCS or detecting sources on an image, to complex combinations of such operations.
+
+**Tasks** are hierarchical.
+Each task may may call other tasks to perform some of its data processing; we say that a **parent task** calls a **subtask**.
+The highest-level task is called the **top-level task**.
+To call a subtask, the parent task constructs the subtask and then calls methods on it.
+Thus data transfer between tasks is simply a matter of passing the data as arguments to function calls.
```

Comment:
Tasks are the fundamental level of code reuse, and we should be designing things that way.  This is _why_ the run method is required to take objects, not dataRefs as programmers usually has the former lying around but not the latter.

Think about the reuse of tasks from Notebooks -- all tasks should be usable that way.   Some tasks written as `CmdLineTask` (e.g. ISRTask) are useful from Notebooks too, which is why the Task and CmdLineTask rules are the same even though the latter, when run from the command line, happen to have a `dataRef` available.